src/link/link_info/gre/gre_tun.rs

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: MIT
+
+use std::net::Ipv4Addr;
+
+use crate::ip::parse_ipv4_addr;
+use anyhow::Context;
+use byteorder::{ByteOrder, NetworkEndian};
+use netlink_packet_utils::{
+    nla::{DefaultNla, Nla, NlaBuffer},
+    parsers::{parse_u16_be, parse_u32_be, parse_u8},
+    DecodeError, Parseable,
+};
+
+use super::{
+    GreEncapFlags, GreEncapType, GreIOFlags, IFLA_GRE_COLLECT_METADATA,
+    IFLA_GRE_ENCAP_DPORT, IFLA_GRE_ENCAP_FLAGS, IFLA_GRE_ENCAP_SPORT,
+    IFLA_GRE_ENCAP_TYPE, IFLA_GRE_FWMARK, IFLA_GRE_IFLAGS, IFLA_GRE_IKEY,
+    IFLA_GRE_LOCAL, IFLA_GRE_OFLAGS, IFLA_GRE_OKEY, IFLA_GRE_PMTUDISC,
+    IFLA_GRE_REMOTE, IFLA_GRE_TOS, IFLA_GRE_TTL, IFLA_GRE_UNSPEC,
+};
+
+pub use InfoGreTun as InfoGreTap;
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[non_exhaustive]
+pub enum InfoGreTun {
+    Unspec,
+    //Link,
```

Comment:
Why comment it out? It is just a u32 interface index.

Reply:
Haven't looked into it. If you say so I will implement it as such.